.dagger/test.go

```diff
@@ -159,6 +163,82 @@ func (t *Test) Specific(
 	)
 }
 
+// Run specific tests
+func (t *Test) Dump(
```

Comment:
nit: do you think it would be possible to instead just continue to centralize around the `test specific` command and make the new args here optional args to that function? Just to limit our number of variations on different functions for running tests.

I guess the trickiness comes from the fact that `test specific` currently errors when the tests fail, but that would prevent us from returning the `Directory` of pprof stuff...

Maybe something like returning a new object like `TestReport` that includes fields for the directory of pprof dumps, whether the tests failed or not, etc.

None of this is a blocker, just throwing it out there as a change we'll likely want at one point or another.